### ISSUE TITLE
fs windows tests doesn't pass when run on Windows

### ISSUE THREAD

**Issue by jsturtevant** [kubernetes/kubernetes#99409]
#### What happened:

It was discovered in #99221 that https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/fs/fs_windows_test.go is ~actually running on Linux (there is no build tag for windows on that file) and therefore doesn't tests windows functionality~ doesn't pass when running on a windows machine.

This should be addressed as part of #51540 which would ensure the windows unit tests are running.

#### What you expected to happen:
files with `_windows_test.go` should test Windows functionality

#### Environment:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by liggitt** [comment]
https://golang.org/cmd/go/#hdr-Build_constraints makes it sound like the _windows suffix auto-tags as windows

> If a file's name, after stripping the extension and a possible _test suffix, matches any of the following patterns:
>
* `*_GOOS`
* `*_GOARCH`
* `*_GOOS_GOARCH`
>
> (example: source_windows_amd64.go) where GOOS and GOARCH represent any known operating system and architecture values respectively, then the file is considered to have an implicit build constraint requiring those terms (in addition to any explicit constraints in the file).

is that not the case?

**Comment by liggitt** [comment]
we have a lot of files using that convention, so if that doesn't work... that's concerning

**Comment by jsturtevant** [comment]
You are right, a bit embarrassing but I had the wrong build tags set up in my environment and was getting the wrong output.  I was able to get the tests to build for windows but they seems to be failing:

```
GOOS=windows go test -c      
ls
BUILD  fs.go  fs.test.exe  fs_unsupported.go  fs_windows.go  fs_windows_test.go
```

Then on a windows machine: 

```
PS C:\projects> .\fs.test.exe
— FAIL: TestDiskUsage (0.00s)
fs_windows_test.go:61: TestDiskUsage failed: CreateFile C:\Users\jstur\AppData\Local\Temp\dir_1076373467/test: The system cannot find the file specified.
FAIL
```

I've opened #99474 to get these to pass on windows.

**Comment by jsturtevant** [comment]
fixed the fs tests in #99221
